dtfs-central-api/src/v1/controllers/tfm/deal-cancellation/tfm-post-deal-cancellation.controller.ts

```diff
@@ -0,0 +1,45 @@
+import { Response } from 'express';
+import { HttpStatusCode } from 'axios';
+import { ApiError, ApiErrorResponseBody, AUDIT_USER_TYPES, CustomExpressRequest, TfmDealCancellationResponse } from '@ukef/dtfs2-common';
+import { validateAuditDetailsAndUserType } from '@ukef/dtfs2-common/change-stream';
+import { TfmDealCancellationRepo } from '../../../../repositories/tfm-deals-repo';
+import { PostDealCancellationPayload } from '../../../routes/middleware/payload-validation/validate-post-deal-cancellation-payload';
+import { shouldDealBeCancelledImmediately } from '../../../../services/deal-cancellation/post-deal-cancellation.service';
+
+type PostTfmDealCancellationRequest = CustomExpressRequest<{
+  reqBody: PostDealCancellationPayload;
+}>;
+
+type PostTfmDealCancellationResponse = Response<ApiErrorResponseBody | TfmDealCancellationResponse>;
+
+/**
+ * Submits the TFM deal cancellation object
+ * @param req - The request object
+ * @param res - The response object
+ */
+export const postTfmDealCancellation = async (req: PostTfmDealCancellationRequest, res: PostTfmDealCancellationResponse) => {
```

Comment:
Unsure if I should call this `postTfmDealCancellation` or `submitTfmDealCancellation`. Any preference?